Modules/MachineLearning/2.0/deploy.json

```diff
@@ -0,0 +1,228 @@
+      "metadata": {
+        "description": "Optional. Machine Learning Plan scale-out capacity of the resource"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Optional. Location for all resources"
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Required. Storage Account Name"
+      }
+    },    
+    "storageAccountSku": {
+      "type": "string",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_RAGRS",
+        "Standard_ZRS",
+        "Premium_LRS",
+        "Premium_ZRS",
+        "Standard_GZRS",
+        "Standard_RAGZRS"
+      ],
+      "defaultValue": "Standard_GRS",
+      "metadata": {
+        "description": "Optional. Storage Account sku type"
+      }
+    },
+    "cuaId": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered"
+      }
+    },
+    "tagEnvironment": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the Environment"
+      }
+    },
+    "tagProject": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the project"
+      }
+    },
+    "tagApplication": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the application"
+      }
+    },
+    "tagOwner": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The business owner for the application"
+      }
+    },
+    "tagOwnerEmail": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The Email address of the business owner for the application"
+      }
+    }
+  },
+  "variables": {
+    "machinelearningName": "[parameters('workspaceName')]",
+    "machinelearningResourceId": "[resourceId('Microsoft.MachineLearning/workspaces', parameters('workspaceName'))]",
+    "storageAccountResourceId": "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]",
+    "pidName": "[concat('pid-', parameters('cuaId'))]"
+  },
+  "resources": [
+    {
+      "condition": "[not(empty(parameters('cuaId')))]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2018-02-01",
+      "name": "[variables('pidName')]",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": []
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Storage/storageAccounts",
```

Comment:
Same as above, this resource is not related to Machine Learning, instead Machine Learning uses a Storage Account, since we already have a Storage Accounts module, let's pass the Storage Account Id instead.

Reply:
Removed and StorageResourceId Parameter now required